### ISSUE TITLE
[Feature] Redirect common 'bang' directly inside the app

### ISSUE THREAD

**Issue by gwenzek** [duckduckgo/android#221]
For now when I use '!gm' inside duckduckgo, my browser is opened (I use an external browser) with the duckduckgo url. Then the browser gets redirected to a Google maps url, which is opened with Maps according to my settings.

It would be awesome if the !gm search was converted inside the DuckDuckGo app into a Google Maps url, which could be directly opened with the corresponding app. That would avoid launching the browser for nothing.

This could be extended to frequent bang shortcuts.

**Comment by maheswaranapk** [comment]
For common 'bang', Shall we have a SnackBar at Bottom which have an option to open the App.

![device-2016-05-26-201646_1](https://cloud.githubusercontent.com/assets/5757251/15579758/99f6b0e4-2383-11e6-80c3-e8f06a548543.png)

On Clicking "Open" will open Google Maps App.

If its ok, I will work it on for most common bangs and I will send a PR.

**Comment by nilnilnil** [comment]
It sounds like the reason that's happening is because you're using an external browser. Is that right? Does this happen for the default case? /cc @moollaza @alohaas

**Comment by moollaza** [comment]
@maheswaranapk thanks for posting this suggestion!

I have never used an external browser, but I just tried and I see now that the DDG send the link to Chrome which then recognizes the Maps URL and opens the native app.

Without the external browser option enabled, !gm shows a web result for Google Maps, within the DDG App browser.

It sounds like identifying if URLs can be resolved by a local app would be useful regardless of whether or not you have the app set to use an external browser. If the user does I image we'd just send them directly to that app, and if they don't I think prompting them to open the result in the native app would be great.

@nilnilnil would that be feasible?

**Comment by gwenzek** [comment]
I usually use Firefox or Chrome but I also tested with the embedded browser and it also open Google maps inside the browser (inside the Duckduckgo app this time). I'd prefer '!m' to allow me to search inside the Google Maps app directly.

**Comment by nilnilnil** [comment]
I think this would be reasonable as a setting, but not default. If the setting was set, we could have the system fire the intent instead of opening the browser internally.

This would have to consider the external browser setting too.

@fgei @sreilly what do you think?

**Comment by fgei** [comment]
yes, I agree that an option would be nice.
we could make something like an "Use External App" setting, and if it is enabled we check if we can resolve the intent with another app and launch it

**Comment by nilnilnil** [comment]
@maheswaranapk is that something you're interested in working on?

**Comment by maheswaranapk** [comment]
@nilnilnil  Yes, Having "Use External App" Option in Settings is a good Choice.

**Comment by edgesince84** [comment]
hey @maheswaranapk I wanted to see if you still planned on finishing this up - we really appreciate your contribution and want to see this go live. We're closing up old PRs and issues in the next week, so let me know if I should leave this open for you!

If you'd rather work on something new, we'd like to invite you to our [new forum](https://forum.duckduckhack.com/) to preview how we're improving programming-related searches. There are a lot of important tasks up for grabs that we'd like your help with, so feel free to and [jump in](https://forum.duckduckhack.com/t/duckduckhack-programming-mission-overview/53).

**Comment by maheswaranapk** [comment]
hey @edgesince84, I had done something but I failed to send the PR. Please keep this issue open and I will send an PR during this weekend.

**Comment by edgesince84** [comment]
thanks @maheswaranapk!  glad to see you tackling this some more 😄
